server-state-machine.go

```diff
@@ -239,10 +239,6 @@ func (state serverStateStart) Next(hr handshakeMessageReader) (HandshakeState, [
 			logf(logTypeHandshake, "[ServerStateStart] Error in PSK negotiation [%v]", err)
 			return nil, nil, AlertInternalError
 		}
-		if clientSentCookie && initialCipherSuite.Suite != params.Suite {
```

Comment:
Why do you think this check is not necessary?

>  Servers MUST ensure that they negotiate the same cipher suite when receiving a conformant updated ClientHello

Are you assuming that this happens automatically?

Reply:
I believe it is done here: https://github.com/bifurcation/mint/blob/master/server-state-machine.go#L253